crates/sargon/src/system/sargon_os/transactions/sargon_os_transaction_analysis.rs

```diff
@@ -0,0 +1,705 @@
+    ) -> Result<TransactionToReview> {
+        self.perform_transaction_preview_analysis(
+            instructions,
+            blobs,
+            message,
+            is_wallet_transaction,
+            nonce,
+            notary_public_key,
+        )
+        .await
+    }
+}
+
+impl SargonOS {
+    pub async fn perform_transaction_preview_analysis(
+        &self,
+        instructions: String,
+        blobs: Blobs,
+        message: Message,
+        is_wallet_transaction: bool,
+        nonce: Nonce,
+        notary_public_key: PublicKey,
+    ) -> Result<TransactionToReview> {
+        let network_id = self.profile_state_holder.current_network_id()?;
+        let gateway_client = GatewayClient::new(
+            self.clients.http_client.driver.clone(),
+            network_id,
+        );
+        let transaction_manifest =
+            TransactionManifest::new(instructions, network_id, blobs)?;
+
+        // Get the transaction preview
+        let transaction_preview = self
+            .get_transaction_preview(
+                gateway_client,
+                transaction_manifest.clone(),
+                network_id,
+                message,
+                nonce,
+                notary_public_key,
+            )
+            .await?;
+        let engine_toolkit_receipt = transaction_preview
+            .radix_engine_toolkit_receipt
+            .ok_or(CommonError::FailedToExtractTransactionReceiptBytes)?;
+
+        // Analyze the manifest
+        let execution_summary = transaction_manifest
+            .execution_summary_with_engine_toolkit_receipt(
+                engine_toolkit_receipt,
+            )?;
+
+        // Transactions created outside of the Wallet are not allowed to use reserved instructions
+        if !is_wallet_transaction
+            && !execution_summary.reserved_instructions.is_empty()
+        {
+            return Err(
+                CommonError::ReservedInstructionsNotAllowedInManifest {
+                    reserved_instructions: execution_summary
+                        .reserved_instructions
+                        .iter()
+                        .map(|i| i.to_string())
+                        .collect(),
+                },
+            );
+        }
+
+        Ok(TransactionToReview {
+            transaction_manifest,
+            execution_summary,
+        })
+    }
+
+    async fn get_transaction_preview(
+        &self,
+        gateway_client: GatewayClient,
+        manifest: TransactionManifest,
+        network_id: NetworkID,
+        message: Message,
+        nonce: Nonce,
+        notary_public_key: PublicKey,
+    ) -> Result<TransactionPreviewResponse> {
+        let signer_public_keys =
+            self.extract_transaction_signers(manifest.summary()).await?;
+        let epoch = gateway_client.current_epoch().await?;
+        let header = TransactionHeader::new(
+            network_id,
+            epoch,
+            Epoch::from(epoch.0 + 10),
```

Comment:
`10` should probably be a `pub const` on `Epoch` it self:

```rust
impl Epoch {
    /// Cirka 1 hour, since one epoch is cirka 6 minutes.
    pub const DEFAULT_EPOCH_WINDOW_SIZE: u64 = 10;
}
```

And perhaps a pub ctor:

```rust
impl Epoch {
   pub fn window_end_from_start(start: Self) -> Self {
       Self::new(start.0 + Self::DEFAULT_EPOCH_WINDOW_SIZE);
   }
}
```